griptape/tools/email_client/tool.py

```diff
@@ -160,27 +167,43 @@ def send(self, params: dict) -> InfoArtifact | ErrorArtifact:
         smtp_host = self.smtp_host
         smtp_port = int(self.smtp_port)
 
-        to_email = input_values["to"]
-        subject = input_values["subject"]
-        msg = MIMEText(input_values["body"])
+        # Create a multipart email and set headers
+        msg = MIMEMultipart()
+        msg["From"] = smtp_user
+        msg["To"] = input_values["to"]
+        msg["Subject"] = input_values["subject"]
+
+        # Attach the body to the email
+        msg.attach(MIMEText(input_values["body"]))
+
+        # Process attachments
+        attachments = input_values.get("attachments", [])
+        for file_path in attachments:
+            try:
+                with open(file_path, "rb") as attachment:
+                    part = MIMEBase("application", "octet-stream")
+                    part.set_payload(attachment.read())
+                    encoders.encode_base64(part)
+                    part.add_header(
+                        "Content-Disposition",
+                        f"attachment; filename= {file_path.split('/')[-1]}"
+                    )
+                    msg.attach(part)
+            except Exception as e:
+                logging.error(f"Error attaching {file_path}: {e}")
```

Comment:
Should we fail the whole activity if any of the files can't be attached? Not actually sure what the right behavior should be but leaning towards returning an `ErrorArtifact` here.

Reply:
Done!

Reply:
Ha, I wasn't sure if that's the right thing to do. What do you think?

@collindutter, @andrewfrench?

Reply:
Sorry very late to this comment 🙃. I don't know if this should always be the case, but I think here it makes sense to return an Error Artifact if any fail to attach. If I wanted to send an email with important documents and only some of them got attached, I would consider that a failure.

Reply:
I agree. Let's fail the whole activity in this case.